cmd/talosctl/pkg/talos/artifacts/copy.go

```diff
@@ -0,0 +1,84 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package artifacts handles manifest traversal for Overalys and Extensions.
+package artifacts
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/siderolabs/go-retry/retry"
+)
+
+// Mirror mirrors the given images to the destination registry.
+func Mirror(ctx context.Context, options []crane.Option, images []string, destRegistry string) error {
+	// TODO:
```

Comment:
TODO needs to be done for future?

Reply:
I want to copy signatures in this PR, rest will be in the following RPs.